### ISSUE TITLE
[BUG] When toggle is turned on or off it must notify the user of the new state

### ISSUE THREAD

**Issue by acovher2** [rokwire/illinois-app#5751]
**Describe the bug**
When toggle is turned on or off it must notify the user of the new state. 

WCAG requires that toggles provide clear, non-visual feedback on their state (on/off or pressed/not pressed) to screen reader users, violating accessibility if only a sound plays. A toggle turning "off" with only a sound fails to inform users of the new state, requiring descriptive text (e.g., "Toggle off") to meet WCAG standards.

[WCAG 4.1.2 Name, Role, Value ](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)is violated. The component must announce its name, role (button/toggle), and state (on/off) when changed. 

**To Reproduce**
_Using TalkBack or VoiceOver_
1. Go to 'Settings'
2. In the dropdown choose  'My food filter'
3. Gesture through to a toggle, 'Halal'
4. "Off, switched off, double tap to switch halal on, switch" is announced by the screen reader.

**Expected behavior**
The user is notified that the switch is in off position, but when user double taps to switch on, they are notified that it has been switched on by assistive technologies(AT). Any subsequent state changes to the same toggle button are also announced by AT. 

**Actual behavior**
The user is notified that the switch is in off position, but when user double taps to switch on, they are **not** notified that it has been switched on by assistive technologies.   Any subsequent state changes to the same toggle button are **not** announced by AT. 

**Screenshots**

https://github.com/user-attachments/assets/aba65eba-2765-450a-94a9-6143441a5610

**Smartphone (please complete the following information):**
- Device: Google Pixel 8a
 - OS:  Android 16
 - Version 7.3.50

 - Device: iPhone 14+
 - OS:  18.6.2
 - Version 7.3.48

![Image](https://github.com/user-attachments/assets/7e5258ce-8809-42b1-9b78-0b0f14399db6)

![Image](https://github.com/user-attachments/assets/4deb3837-2df4-4e69-b0b6-25c29a06689f)

**Additional context**
This should be the functionality where any toggles/checkboxes in the entire app. I will list subsequent examples of this violation in this ticket.

**Comment by acovher2** [comment]
Another toggle that is affected by this issue is the favorites button almost everywhere in app. It is not announcing whether it is checked or unchecked when it gains focus, and not announcing subsequent state changes either. See video. 

https://github.com/user-attachments/assets/6141f5f7-48c7-4305-912d-f159153c0a73
